accepted/00101-make_cve_reports_possible.md

```diff
@@ -0,0 +1,60 @@
+- Feature Name: make_cve_reports_possible
+- Start Date: 2024-04-12 
+
+# Summary
+[summary]: #summary
+
+Create a meaninful CVE Audit that would ran as a report and it would be sorted per CVE-Number
+
+# Motivation
+[motivation]: #motivation
+
+- Why are we doing this? The CVE Audit is static and does not provide an overview from the hosts affected by a CVE.
+- What use cases does it support? Produce real-time reports sorted by CVE-Number for the auditors and security teams
+- What is the expected outcome? Have a table with the CVE Numbers and what hosts are affected. Have a report that could be called via REST-API and also could be integrated on every CI/CD workflows from operational teams.
+
+We have a static and not a understable overview from the hosts affected by a CVE. As today, one should open the CVE Audit option on the menu, search for a CVE number, add to the form and expect a list from hosts affected by only this CVE. If there is a higher number from CVEs would be a lot of working doing a manual search per CVE Number. This makes the life from infrastructure managers so difficult -it is not possible to have an overview that is always updated and showed in the dashboard -
```

Comment:
Can you please elaborate on what's not understandable from the current overview? 

How do you envision it from UI perspective? Available CVE number is huge.

Would you like to identify which system is affected by a particular CVE(s), or if any system is affected by any CVE at all? Are you solely interested in CVEs that affect systems or even in those CVEs which doesn't affect any of  the onboarded system?

Currently, we have limitation that we could only tell if a system is affected by CVE if assigned channels has patch which in turn fix that CVE. Depending on if that patch has been applied or not, we tell if system is affected or not. Based on this, we already show on system overview page if systems has some security updates or all good. 


Now if I summarize and IIUC, what you likely want is the ability to provide a list of CVEs and then receive results indicating whether systems are affected or not based on that list. Please correct me, if that wouldn't suffice.

Reply:
Hi @admd !

Can you please elaborate on what's not understandable from the current overview?
It is everything static. How a admin will know if there is a new CVE affecting their servers and what CVE number it is? I did a lot of demos for customers showing the CVE Audit and I always had to search for a CVE number and check if one of the hosts is affected. Similar to NeuVector[0] we could use the same databases which they use to do this CVE checks and show it the hosts from Uyuni are affected. 

It is nice that the CVEs are just show when the channel has a patch against it - but would not make more sense to display the real situation and what patches are being delivered? If it is just based on the channels from Uyuni it would give a false positive - well, my channels do not have the patch for CVE 12345, it means that my hosts are not affected. 

[0] - Example how CVE Databases works for NeuVector: https://open-docs.neuvector.com/scanning/updating